features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/FlagsClient.kt

```diff
@@ -77,109 +108,351 @@ interface FlagsClient {
+         * Gets the [FlagsClient] with the specified name from the SDK core.
          *
-         * This method is thread-safe and will return the same client instance for the same SDK core
-         * across multiple calls. If no client has been registered for the given SDK core, a no-op
-         * implementation will be returned instead.
+         * If no [FlagsClient] exists with the given name, returns a [NoOpFlagsClient] that logs
+         * critical errors but never crashes.
          *
-         * @param sdkCore the [SdkCore] instance to retrieve the client for. If not provided,
-         * the default Datadog SDK instance will be used.
-         * @return the [FlagsClient] associated with the given SDK core, or a no-op client.
-         * if no client is registered for this SDK core.
+         * @param name the [FlagsClient] name. Defaults to "default".
+         * @param sdkCore the SDK instance. Defaults to the default Datadog instance.
+         * @return the [FlagsClient] with the specified name, or [NoOpFlagsClient] if not found.
          */
         @JvmOverloads
         @JvmStatic
-        fun get(sdkCore: SdkCore = Datadog.getInstance()): FlagsClient = synchronized(registeredClients) {
-            val client = registeredClients[sdkCore]
-            if (client == null) {
-                val errorMsg = "No FlagsClient for the SDK instance with name ${sdkCore.name} " +
-                    "found, returning no-op implementation."
-                (sdkCore as? FeatureSdkCore)
-                    ?.internalLogger
-                    ?.log(
-                        InternalLogger.Level.WARN,
-                        InternalLogger.Target.USER,
-                        { errorMsg }
+        fun get(name: String = DEFAULT_CLIENT_NAME, sdkCore: SdkCore = Datadog.getInstance()): FlagsClient {
+            val key = ClientKey(sdkCore, name)
+
+            synchronized(registeredClients) {
+                val client = registeredClients[key]
+
+                if (client == null) {
+                    val logger = (sdkCore as? FeatureSdkCore)?.internalLogger
+
+                    // Log at get() level for visibility
+                    logger?.log(
+                        InternalLogger.Level.ERROR,
+                        listOf(InternalLogger.Target.USER, InternalLogger.Target.MAINTAINER),
+                        {
+                            "No FlagsClient with name '$name' exists for SDK instance '${sdkCore.name}'. " +
+                                if (name == DEFAULT_CLIENT_NAME) {
+                                    "Create a client first using: FlagsClient.Builder().build(). "
+                                } else {
+                                    "Create a client first using: FlagsClient.Builder(\"$name\").build(). "
+                                } +
+                                "Returning NoOpFlagsClient which always returns default values."
+                        }
```

Comment:
minor: just to make it easier to read

```suggestion
                if (client == null) {
                    val logger = (sdkCore as FeatureSdkCore).internalLogger
                    val clientBuildStatement = if (name == DEFAULT_CLIENT_NAME) {
                        "Create a client first using: FlagsClient.Builder().build(). "
                    } else {
                        "Create a client first using: FlagsClient.Builder(\"$name\").build(). "
                    }

                    logger.log(
                        InternalLogger.Level.ERROR,
                        listOf(InternalLogger.Target.USER, InternalLogger.Target.MAINTAINER),
                        {
                            "No FlagsClient with name '$name' exists for SDK instance '${sdkCore.name}'. " +
                                " $clientBuildStatement. Returning NoOpFlagsClient which always returns default values."
                        }
```